### ISSUE TITLE
vagrant 1.6.5+ docker 1.6.1 on Debian Testing does not seem to work

### ISSUE THREAD

**Issue by rahul-amaram-vizury** [tmatilai/vagrant-proxyconf#125]
Here is the configuration:

```ruby
Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  # All Vagrant configuration is done here. The most common configuration
  # options are documented and commented below. For a complete reference,
  # please see the online documentation at vagrantup.com.

  config.proxy.http     = "http://172.17.42.1:3128/"
  config.proxy.https    = "http://172.17.42.1:3128/"
  config.proxy.ftp      = "http://172.17.42.1:3128/"
  config.proxy.no_proxy = "localhost,127.0.0.1"
  ...
  ...
  ...
  config.vm.provider "docker" do |d|
    d.build_dir = "vagrant"
    d.has_ssh = true
  end
  ...
  ...
  ...
end
```

Is anything else needed? When I run vagrant using --debug , it shows that the proxyconf plugin got loaded. However, the proxy configuration does not seem to be used by docker. I believe I also restarted docker though I am not sure if this is really needed.

**Comment by mrsheepuk** [comment]
I think this is caused by needing to set the proxy config in /etc/systemd/system/docker.service.d/http-proxy.conf on the newer deb/ubuntu versions which use systemd instead of upstart.

I've managed to get docker respecting the proxy by manually adding that file with content like:

    [Service] 
    Environment="HTTP_PROXY=http://10.0.0.10:8080/" "HTTPS_PROXY=http://10.0.0.10:8080/"

... however I don't know Ruby so my attempts to update vagrant-proxyconf to push this out have been unsuccessful so far.

**Comment by mrsheepuk** [comment]
I've made a pull request #133 which may address this, it works for me but as I say I'm not an experienced Ruby person so I may have made some fundamental errors in it!

**Comment by otahi** [comment]
@rahul-amaram-vizury I am sorry, I could not respond. Please tell me the Debian version.
@mrsheepuk Thank you for reminding me, I will check it later.

**Comment by rahul-amaram-vizury** [comment]
I don't remember the complete context for this ticket as it has been a long time. I am currently using Debian Testing (stretch). I think I had implemented some workaround for this problem but I'm sorry I cannot recollect it now.

**Comment by otahi** [comment]
OK. Ubuntu/Debian Docker official package issue is discussed on #133.
So I have closed this issue.

**Comment by xliiv** [comment]
Is this solved because having ubuntu 18.04 as guest i still to manually change it like here:
https://docs.docker.com/config/daemon/systemd/#httphttps-proxy
